lib/pure/osproc.nim

```diff
@@ -1387,22 +1388,26 @@ elif not defined(useNimRtl):
             raiseOsError(osLastError())
           else:
             # Continue waiting if needed
-            if getTime() >= deadline:
+            if getMonoTime() >= deadline:
               # Previous version of `waitForExit`
               # foricibly killed the process.
               # We keep this so we don't break programs
               # that depend on this behavior
               if posix.kill(p.id, SIGKILL) < 0:
                 raiseOSError(osLastError())
             else:
-              let newWait = getTime() + delay
+              const max = 999_999_999
+              let 
+                newWait = getMonoTime() + delay
+                ns = newWait.ticks() mod max
```

Comment:
Why `mod 999_999_999` and not `mod 1_000_000_000`?

Reply:
I had just woken up :sweat_smile: